File: src/archive/CSESJava/src/main/java/org/allan7yin/searching/CollectingNumbers.java
package org.allan7yin.searching;
import java.util.Scanner;

public class CollectingNumbers {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        int[] nums = new int[n + 1];

        for (int i = 0; i < n; i++) {
            int num = scanner.nextInt();
            nums[num] = i;
        }

        int count = 1;
        for (int i = 2; i <= n; i++) {
            if (nums[i] < nums[i - 1]) {
                count++;
            }
        }

        System.out.println(count);

        scanner.close();
    }
}

File: src/archive/CSES/Sorting_searching/collecting_numbers.cpp
#include <iostream>
#include <utility>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> nums(n + 1, 0);

    for (int i = 0; i < n; i++) {
        int num;
        cin >> num;
        nums[num] = i;
    }

    int count = 1;
    for (int i = 2; i <= n; i++) {
        if (nums[i] < nums[i - 1]) {
            count++;
        }
    }

    cout << count << endl;
}
